File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <ctype.h>

bool is_pangram(const char *sentence) {
    if (!sentence) 
        return false;
    
    const char abc[] = "abcdefghijklmnopqrstuvwxyz";

    for (const char *i = abc; *i; i++) {
        bool found = false;

        for (const char *j = sentence; *j; j++) {
            char tmp = *j;
            if (tmp >= 'A' && tmp <= 'Z')
                tmp += 32;
            
            if (tmp == *i) {
                found = true;
                break;
            }
        }

        if (!found)
            return false;
    }

    return true;
}

File: solutions/ruby/pangram/1/pangram.rb
module Pangram
  def self.pangram?(sentence)
    counts = Hash.new(0)

    sentence
      .downcase
      .chars
      .map { |c| counts[c] += 1 if c >= 'a' && c <= 'z' }

    counts.keys.size == 26
  end
end